inc/health-check-curl-version.php

```diff
@@ -29,9 +29,9 @@ public function run() {
 
 		if ( ! $this->is_my_yoast_api_reachable() && $this->is_recent_curl_version() ) {
 			$this->label = sprintf(
-				/* translators: %1$s expands to 'my.yoast.com'. */
-				esc_html__( 'Your site can not connect to %1$s', 'wordpress-seo' ),
-				'my.yoast.com'
+				/* translators: %1$s expands to 'Yoast'. */
```

Comment:
the change is minimal, but I don't understand how is_my_yoast_api_reachable and is_recent_curl_version  concludes that premium plugins cannot update. Sounds like we need an is_premium_plugin function instead.